File: app/src/main/java/com/artem/tusaandroid/NativeLibrary.java
package com.artem.tusaandroid;

import android.content.res.AssetManager;

public class NativeLibrary {
    static {
        System.loadLibrary("tusaandroid");
    }
    public static native void render();
    public static native void onSurfaceChanged(int width, int height);
    public static native void surfaceCreated(AssetManager assetManager);
    public static native void noOpenGlContextInit(AssetManager assetManager, float appStartScaleFactor);
    public static native void setupCacheDirAbsolutePath(String path);
    public static native void drag(float dx, float dy);
    public static native void scale(float scaleFactor);
    public static native void doubleTap();
    public static native void onStop();

    public static native void onDown();
}

File: app/src/main/java/com/artem/tusaandroid/NativeLibrary.kt
package com.artem.tusaandroid

import android.content.res.AssetManager
import com.artem.tusaandroid.location.CameraPos

object NativeLibrary {
    init {
        System.loadLibrary("tusaandroid")
    }

    external fun create()
    external fun cleanup()

    external fun render()
    external fun onSurfaceChanged(width: Int, height: Int)
    external fun surfaceCreated(assetManager: AssetManager?)
    external fun surfaceDestroyed()
    external fun noOpenGlContextInit(assetManager: AssetManager?, requestTile: RequestTile)
    external fun drag(dx: Float, dy: Float)
    external fun scale(scaleFactor: Float): Float
    external fun doubleTap()
    external fun onDown()
    external fun setCameraPos(latitude: Float, longitude: Float, zoom: Float)
    external fun getCameraPos(): CameraPos
    external fun confirmedClick(x: Float, y: Float, ignore: Long): MapClickResult
    external fun deselectSelectedMarker()

    // Markers
    external fun removeMarker(key: Long)
    external fun updateMarkerAvatar(key: Long, avatarBuffer: ByteArray)
    external fun updateMarkerGeo(key: Long, latitude: Float, longitude: Float)
    external fun addMarker(key: Long, latitude: Float, longitude: Float, avatarBuffer: ByteArray)
    external fun existMarker(key: Long): Boolean
}